src/transformState.js

```diff
@@ -5,7 +5,32 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (true) {
+      case action.type === 'addProperties':
+        for (const key in action.extraData) {
+          state[key] = action.extraData[key];
+        }
+        break;
+
+      case action.type === 'removeProperties':
+        const actionKeysToRemove = action.keysToRemove;
+
+        for (let i = 0; i < actionKeysToRemove.length; i++) {
+          delete state[action.keysToRemove[i]];
+        }
+        break;
+
+      case action.type === 'clear':
+        for (const keyState in state) {
+          delete state[keyState];
+        }
+        break;
+
+      default:
+        return;
```

Comment:
Here you can throw some error to inform user that something went wrong:
```suggestion
        throw Error(`Action type ${action.type} is not found`);
```